server/neptune/gateway/event/comment_handler.go

```diff
@@ -72,31 +96,59 @@ func NewCommentEventWorkerProxy(
 			globalCfg:        globalCfg,
 		},
 		neptuneWorkerProxy: &NeptuneWorkerProxy{
-			logger:         logger,
-			signaler:       signaler,
-			commentCreator: commentCreator,
+			logger:             logger,
+			signaler:           signaler,
+			commentCreator:     commentCreator,
+			requirementChecker: requirementChecker,
 		},
 		vcsStatusUpdater:  vcsStatusUpdater,
 		rootConfigBuilder: rootConfigBuilder,
+		errorHandler:      errorHandler,
 	}
 }
 
 type NeptuneWorkerProxy struct {
-	logger         logging.Logger
-	signaler       deploySignaler
-	commentCreator commentCreator
+	logger             logging.Logger
+	signaler           deploySignaler
+	commentCreator     commentCreator
+	requirementChecker requirementChecker
 }
 
 func (p *NeptuneWorkerProxy) Handle(ctx context.Context, request *http.BufferedRequest, event Comment, cmd *command.Comment, roots []*valid.MergedProjectCfg) error {
-	// currently the only comments on platform mode are force applies, we can add to this as necessary.
-	if !cmd.ForceApply {
+	// currently the only comments on platform mode are applies, we can add to this as necessary.
+	if cmd.Name != command.Apply {
 		return nil
 	}
 
+	triggerInfo := workflows.DeployTriggerInfo{
+		Type:  workflows.ManualTrigger,
+		Force: cmd.ForceApply,
+	}
+
 	platformModeRoots := partitionRootsByMode(valid.PlatformWorkflowMode, roots)
 
-	// let's only comment on the PR if we're fully on platform mode, otherwise there will be duplicates from the legacy worker and this.
-	if len(platformModeRoots) == len(roots) {
+	if cmd.IsForSpecificProject() {
```

Comment:
Are we still supporting directories/workspaces? If not, wouldn't running a specific project be the same as running on one root?

Reply:
yeah its the same as running one root which can be specified using `-p` we don't wanna remove that functionality